PencilChiselCode/Bonfire.cs

```diff
@@ -208,11 +227,13 @@ public void ResetPenumbra()
 
     protected override void Update(GameTime gameTime)
     {
-        if (Controls.JustPressed(ControlKeys.DEBUG))
+        if (Controls.JustPressed(ControlKeys.Debug))
         {
-            DebugMode = !DebugMode;
+            DebugMode = (DebugMode + 1) % 3;
         }
+
         ScreenManager.Update(gameTime);
+        MouseValues.Update();
         Controls.Update();
```

Comment:
Test adding another another `ScreenManager.Update()` after `Controls.Update();`, if that makes game play same as upstream then I would merge with that and fix it properly after..